tcplistener.go

```diff
@@ -148,6 +147,18 @@ func handleListenedConn(address string, conn *net.TCPConn, headerByteSize int, m
 	headerBuffer := make([]byte, headerByteSize)
 	dataBuffer := make([]byte, maxMessageSize)
 	for {
+		// Before we try reading a message, check to see if we've been shutdown or not
+		// That way, shutting down will allow any current message to finish, but block
+		// any receives on the connection after it occurs, in addition to blocking new
+		// connections from being accepted, above in the initial connection code.
+		select {
+		case <-sdChan:
+			conn.Close()
+			return
+		default:
+			// Nothing, continue on to accept incoming bytes
+		}
+
```

Comment:
We talked about this approach offline, and I agree this is actually a much slicker way to do what I'm doing with the select statement inside the listener. The conn will close and in turn, the next read from the conn will be an error, and exit from the handler goroutine.

Will tweak my approach and re-test